### DISCUSSION TITLE
[Simulation question]: SBS Gvalues - Creator model undefined

### DISCUSSION THREAD

**Discussion by JoanaSequeiraAntunes** [topas-nbio/TOPAS-nBio#38] [Simulation-question] [answered]
### Simulation question

Hello everyone,

I am working on calculating G-values for a water cube of 2x2x2 µm³ using a phase space file as the radiation source at the cube's center. Regarding the physics and chemistry, I used as basis the example _GValuesDNADefault.txt_, where the modules _g4em-dna_ and _g4em-dna-chemistry_ were used. 

However, when I run the simulation, I get the following error:

```
-------- EEEE ------- G4Exception-START -------- EEEE -------
*** G4Exception : ITManager002
      issued by : G4ITManager::FindNearest
Bad request : no node found in the IT you are searching closest neighbourg for
*** Fatal Error In Argument *** core dump ***
G4WT2 > G4Track (0x7f3b9aa02800) - track ID = -272, parent ID = -170
G4WT2 >  Particle type : H_2 - creator process : H2O_DNAMolecularDecay, creator model : Undefined
G4WT2 >  Kinetic energy : 0.0385195 eV  - Momentum direction : (-0.0655874,0.662549,0.746142)
G4WT2 >  Step length : 6.23432e-297 fm  - total energy deposit : 1.64791 eV 
G4WT2 >  Pre-step point : (2.26942e-05,7.7441e-06,-1.82046e-05) - Physical volume : target (G4_WATER)
G4WT2 >  - defined by : not available
G4WT2 >  Post-step point : (2.26942e-05,7.7441e-06,-1.82046e-05) - Physical volume : target (G4_WATER)
G4WT2 >  - defined by : e-_G4DNAElectronSolvation - step status : 4
G4WT2 >  *** Note: Step information might not be properly updated.
G4WT2 > 
-------- EEEE -------- G4Exception-END --------- EEEE -------
```

I have attached the parameter file for reference. 
[gvalues_Error.txt](https://github.com/user-attachments/files/17062000/gvalues_Error.txt)

Does anyone know how to fix it?
Thank you in advance for your time and attention. 

Best regards,
Joana Antunes

### TOPAS-nBio category

Chemistry

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by tmasilela** [comment]
Hi Joana,

I have attached a revised version of your parameter file. Which I've tested and runs fine (since I don't have your phase space file I just added 1 MeV electrons as your source). Some things to note:

- **L49** (_d:So/phsp/MeanEnergy = 0.235713413 MeV_) should be removed. While the parameter MeanEnergy exists I don't think it can be used in this context. Furthermore "phsp" is not defined anywhere else in your parameter file.

- **L51** (_d:Sc/GValueInTarget/KillPrimaryIfEnergyLossExceeds = ..._) has historically been used to kill the track to avoid excessive simulation times. You'll notice that in the example this is set to 100% of the beam energy since the volume is small enough (um size) that the simulations finish sufficiently fast. I'm not sure how exactly the use of this parameter with a phase space source works, but unless you specifically want to use it I would just set this parameter to a value higher than the maximum energy within your phase space.

- **L53** (_d:Sc/GValueInTarget/AbortEventIfPrimaryEnergyLossExceeds = ..._) change the units here to match L51

Hope that helps, and good luck with the simulations.

All the best,
Thongchai
[gvalues_Error_EDIT.txt](https://github.com/user-attachments/files/17063906/gvalues_Error_EDIT.txt)

**Reply by JoanaSequeiraAntunes** [reply]
Hi Thongchai, 

Thank you for your quick answer. 
I did what you suggested, but I still get the same error. 
Attached is my phase space file.
[Nanoparticle_header.txt](https://github.com/user-attachments/files/17072697/Nanoparticle_header.txt)
[Nanoparticle_phsp.txt](https://github.com/user-attachments/files/17072698/Nanoparticle_phsp.txt)
(I had to change the extension of the files, so I can upload them).

Best, 
Joana

**Reply by tmasilela** [reply]
Hi Joana,

Do you get the exact same error as in your original post or is it something slightly different?

From my side, In the file I sent "gvalues_Error_EDIT.txt"I commented out L25-32, and removed comments from L19-23 (to include the phase space instead of the beam source) and everything seems to run fine. 

If you can, please post the entire log output of when you try to run the simulation so I can have a bit more information while trying to fix your issue.

All the best,
Thongchai

**Reply by JoanaSequeiraAntunes** [reply]
Hi, 

Here are all the files that I used to run the simulation that gave the output attached.  
[gvalues.txt](https://github.com/user-attachments/files/17145545/gvalues.txt)
[Nanoparticle_header.txt](https://github.com/user-attachments/files/17145546/Nanoparticle_header.txt)
[Nanoparticle_phsp.txt](https://github.com/user-attachments/files/17145547/Nanoparticle_phsp.txt)
[output.txt](https://github.com/user-attachments/files/17145548/output.txt)

Each time I run it, I got the same error, but with slight differences.

Thank you in advance for your time and attention.

Best regards,
Joana Antunes

**Comment by tranngochoang** [comment, marked as answer]
Dear, 

It seems to me the issue comes from FindNearest function where it finds a invalid node. Can you please try to copy content of these attached files to the files G4KDNode.icc (source/processes/electromagnetic/dna/management/include/G4KDNode.icc) and G4KDTree.icc (source/processes/electromagnetic/dna/management/include/G4KDTree.icc) respectively (I cannot attach the original names to this forum). Then, re-compile fully Topas. Then see if the issue still persists ?

[G4KDNode.txt](https://github.com/user-attachments/files/17365428/G4KDNode.txt)
[G4KDTree.txt](https://github.com/user-attachments/files/17365429/G4KDTree.txt)
